application/include/mmul.hpp

```diff
@@ -17,17 +17,17 @@
 using namespace Eigen;
 
 template <typename Executor,
-          typename execution::executor_of_type<inline_executor, Executor> = 0>
+          typename execution::instance_of_base<inline_executor, Executor> = 0>
```

Comment:
I understand your point, the reason the name might be confusing is since I intentionally want to keep It short. This trait will be used everywhere so having a large name will pollute the codebase everywhere.

Reply:
`enable_as_int_if_instance_of_base_t` or `int_if_instance_of_base_t`? I think we need to compromise a little for clarity.

Reply:
Passing a 0 to something is an alternative (and stronger) design for `typename X = default_type` to prevent the user from overloading the design. Once this is known, it's not that confusing. `enable_if_instance_of_base` might be more expressive, but barely saves space (original intent).

> For new API, users demand expressiveness and bold syntax, but once they start using it, they want more concise syntax
- Bjarne (paraphrased)

Apparently, an age old problem